src/net_processing.cpp

```diff
@@ -3169,11 +3175,21 @@ bool PeerLogicValidation::ProcessMessages(CNode* pfrom, std::atomic<bool>& inter
     if (!pfrom->vRecvGetData.empty())
         ProcessGetData(pfrom, chainparams, connman, interruptMsgProc);
 
+    if (!pfrom->orphan_work_set.empty()) {
```

Comment:
Commit 866c805

Not sure if `orphan_work_set` should be guarded by `g_cs_orphans`?

Reply:
No, it's only accessed from the network processing thread like vRecvGetData, so it doesn't need locking.

Ideally it should move to net_processing's structures instead of CNode, along with a bunch of other things, but I wanted to keep the diff small for now.